### PR TITLE
[Common] Add new API to get tensor format of current pad caps

### PR DESCRIPTION
**Pull request by songgot**
- Add gst_tensor_pad_get_format()
- Replace gst_tensor_pad_caps_is_flexible() function to macro.
- Add gst_tensor_pad_caps_is_static and gst_tensor_pad_caps_is_sparse macro

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: gst/nnstreamer/nnstreamer_plugin_api_impl.c [hunk 1]
```diff
@@ -1244,6 +1244,35 @@ gst_tensor_pad_caps_is_flexible (GstPad * pad)
   return ret;
 }
 
+/**
+ * @brief Check current pad caps is sparse tensor.
+ * @param pad GstPad to check current caps
+ * @return TRUE if pad has sparse tensor caps.
+ */
+gboolean
+gst_tensor_pad_caps_is_sparse (GstPad * pad)
```
Comment (jaeyun-jung) [inline gst/nnstreamer/nnstreamer_plugin_api_impl.c:1253]: Discussion: refactoring functions for pad caps
How about implementing new function and removing gst_tensor_pad_caps_is_flexible() / gst_tensor_pad_caps_is_sparse() ?
```
/**
 * @brief Get tensor format of current pad caps.
 * @param pad GstPad to check current caps
 * @return The tensor format of current pad caps. _NNS_TENSOR_FORMAT_END if pad does not have tensor caps.
 */
tensor_format gst_pad_get_tensor_format (GstPad * pad);
```

Comment (songgot) [reply gst/nnstreamer/nnstreamer_plugin_api_impl.c:1253]: okay, Thank you.

Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer_plugin_api_impl.c:1253]: IMO, after implementing the function, we can add simple macro :)
```
#define gst_tensor_pad_caps_is_static(p) (gst_tensor_pad_get_format (p) == _NNS_TENSOR_FORMAT_STATIC)
#define gst_tensor_pad_caps_is_flexible(p) (gst_tensor_pad_get_format (p) == _NNS_TENSOR_FORMAT_FLEXIBLE)
#define gst_tensor_pad_caps_is_sparse(p) (gst_tensor_pad_get_format (p) == _NNS_TENSOR_FORMAT_SPARSE)
```

Comment (songgot) [reply gst/nnstreamer/nnstreamer_plugin_api_impl.c:1253]: Thank you for your kind explanation. :+1:

File: gst/nnstreamer/nnstreamer_plugin_api_impl.c [hunk 2]
```diff
@@ -1216,27 +1216,29 @@ gst_tensor_pad_possible_caps_from_config (GstPad * pad,
...
- */
-gboolean
-gst_tensor_pad_caps_is_flexible (GstPad * pad)
+  * @brief Get tensor format of current pad caps.
+  * @param pad GstPad to check current caps.
+  * @return The tensor_format of current pad caps.
+  *
+  * If pad does not have tensor caps return _NNS_TENSOR_FORMAT_END
+  */
+extern tensor_format
```
Comment (jaeyun-jung) [inline gst/nnstreamer/nnstreamer_plugin_api_impl.c:1225]: plz remove extern in .c

**Review by jaeyun-jung — APPROVED** [review]

**Review by gichan-jang — APPROVED** [review]
LGTM

File: gst/nnstreamer/nnstreamer_plugin_api_impl.c (hunk 2 above)
Comment (songgot) [reply gst/nnstreamer/nnstreamer_plugin_api_impl.c:1225]: okay !!

**Pull request closed without merging**